examples/structured_data/deep_neural_decision_forests.py

```diff
@@ -0,0 +1,454 @@
+print(f"Test dataset shape: {test_data.shape}")
+
+train_data.head().T
+
+"""
+We remove the first record as it is not a valid data example, and we remove a tailing
+'dot' in the class labels.
+"""
+
+test_data = test_data[1:]
+test_data.income_bracket = test_data.income_bracket.apply(
+    lambda value: value.replace(".", "")
+)
+
+"""
+Now we store the train and test data splits locally to CSV files.
+"""
+
+train_data_file = "train_data.csv"
+test_data_file = "test_data.csv"
+
+train_data.to_csv(train_data_file, index=False, header=False)
+test_data.to_csv(test_data_file, index=False, header=False)
+
+"""
+## Define dataset metadata
+
+Here, we define the metadata of the dataset that will be useful for reading and parsing
+the data into input features, and encoding the input features with respect to their types.
+"""
+
+NUMERIC_FEATURE_NAMES = [
+    "age",
+    "education_num",
+    "capital_gain",
+    "capital_loss",
+    "hours_per_week",
+]
+
+CATEGORICAL_FEATURES_WITH_VOCABULARY = {
+    "workclass": list(train_data["workclass"].unique()),
+    "education": list(train_data["education"].unique()),
+    "marital_status": list(train_data["marital_status"].unique()),
+    "occupation": list(train_data["occupation"].unique()),
+    "relationship": list(train_data["relationship"].unique()),
+    "race": list(train_data["race"].unique()),
+    "gender": list(train_data["gender"].unique()),
+    "native_country": list(train_data["native_country"].unique()),
+}
+
+IGNORE_COLUMN_NAMES = ["fnlwgt"]
+
+CATEGORICAL_FEATURE_NAMES = list(CATEGORICAL_FEATURES_WITH_VOCABULARY.keys())
+
+FEATURE_NAMES = NUMERIC_FEATURE_NAMES + CATEGORICAL_FEATURE_NAMES
+
+COLUMN_DEFAULTS = [
+    [0.0] if feature_name in NUMERIC_FEATURE_NAMES + IGNORE_COLUMN_NAMES else ["NA"]
+    for feature_name in CSV_HEADER
+]
+
+TARGET_FEATURE_NAME = "income_bracket"
+
+TARGET_LABELS = [" <=50K", " >50K"]
+
+"""
+## Create tf.data.Dataset for training and evaluation
+
+We create an input function to read and parse the file, and convert features and labels
+into a [`tf.data.Dataset`](https://www.tensorflow.org/guide/datasets)
+for training or evaluation. We also preprocess the input by mapping the target label
+to an index.
+"""
+
+from tensorflow.keras.layers.experimental.preprocessing import StringLookup
+
+taget_label_lookup = StringLookup(
+    vocabulary=TARGET_LABELS, mask_token=None, num_oov_indices=0
+)
+
+
+def get_dataset_from_csv(csv_file_path, num_epochs=None, shuffle=False, batch_size=128):
+    def process(features, target):
+        target_index = taget_label_lookup(target)
+        return features, target_index
+
+    dataset = tf.data.experimental.make_csv_dataset(
+        csv_file_path,
+        batch_size=batch_size,
+        column_names=CSV_HEADER,
+        column_defaults=COLUMN_DEFAULTS,
+        label_name=TARGET_FEATURE_NAME,
+        num_epochs=num_epochs,
+        header=False,
+        na_value="?",
+        shuffle=shuffle,
+    ).map(process)
```

Comment:
You could just do `lambda x, y: (x, target_label_lookup(y))` instead of having a separate function

Reply:
Done